### PR TITLE
Expose legal hold team member status; send client ID with LH event

### PR DESCRIPTION
**Pull request by ChrisPenner**
- [x] Return User legal hold status on members from `GET /teams/:id/members`
- [x] Send Client ID with LH request device event
- [x] SWAGGER

### CODE REVIEW

File: services/galley/src/Galley/Data/LegalHold.hs [hunk 1]
```diff
@@ -73,13 +73,14 @@ selectPendingPrekeys uid =
 dropPendingPrekeys :: MonadClient m => UserId -> m ()
 dropPendingPrekeys uid = retry x5 (write Q.dropPendingPrekeys (params Quorum (Identity uid)))
 
-getUserLegalHoldStatus :: MonadClient m => UserId -> m UserLegalHoldStatus
-getUserLegalHoldStatus uid = do
-    result <- retry x1 (query1 Q.selectUserLegalHoldStatus (params Quorum (Identity uid)))
+getUserLegalHoldStatus :: MonadClient m => TeamId -> UserId -> m UserLegalHoldStatus
+getUserLegalHoldStatus tid uid = do
```
Comment (fisx) [inline services/galley/src/Galley/Data/LegalHold.hs]: If you like, you can drop this function and call `getMember` (or whatever it's called) instead.  Less code, more good!  :-)

(I haven't checked if there is such a more general function, though.)

File: libs/galley-types/src/Galley/Types/Teams.hs [hunk 2]
```diff
@@ -590,6 +606,8 @@ parseTeamMember = withObject "team-member" $ \o ->
     TeamMember <$> o .:  "user"
                <*> o .:  "permissions"
                <*> parseInvited o
+               -- Default to disabled if missing
```
Comment (fisx) [inline libs/galley-types/src/Galley/Types/Teams.hs]: Is that a `Monoid` with overwrite semantics?  Probably, but even more probably that's over-thinking things..

Comment (ChrisPenner) [reply libs/galley-types/src/Galley/Types/Teams.hs]: Do you mean the type itself or the JSON parser?

I'm just using built-in Aeson combinators for specifying a default 🤷‍♂

Comment (fisx) [reply libs/galley-types/src/Galley/Types/Teams.hs]: i meant the type.  If there was a monoid instance then we could say `mempty` here, which takes semantics away from the serialization code.

(not important at all.)

**Pull request merged**